examples/complete/main.tf

```
@@ -0,0 +1,40 @@
+locals {
+  public_cidr_block  = cidrsubnet(var.cidr_block, 1, 0)
+  private_cidr_block = cidrsubnet(var.cidr_block, 1, 1)
+}
+
+module "vpc" {
+  source     = "git::https://github.com/cloudposse/terraform-aws-vpc.git?ref=tags/0.7.0"
+  namespace  = var.namespace
+  stage      = var.stage
+  name       = var.name
+  cidr_block = var.cidr_block
+}
+
+module "public_subnets" {
+  source              = "../../"
+  namespace           = var.namespace
+  stage               = var.stage
+  name                = var.name
+  availability_zones  = var.availability_zones
+  vpc_id              = module.vpc.vpc_id
+  cidr_block          = local.public_cidr_block
+  type                = "public"
+  igw_id              = module.vpc.igw_id
+  nat_gateway_enabled = "true"
```

```suggestion
  nat_gateway_enabled = true
```

Use TF 0.12 first class types eveywhere